cas: add a GetImageManifest function.

sgotti opened the pull request:
Last commit is the interesting one. 

The previous required commit was proposed in #352.

GetImageManifest retrieves the ImageManifest with the specified ImageID from
the AppInfo index.

crawford commented on cas/aciinfo.go:
```diff
@@ -0,0 +1,40 @@
+)
+
+type ACIInfo struct {
+	Im     *schema.ImageManifest
```
Can you make this member name more descriptive?

sgotti replied:
Tried to add a description.

crawford commented on cas/aciinfo.go:
```diff
@@ -0,0 +1,40 @@
+	m.Key = key
+	m.Latest = latest
+	m.Time = t
+	return m
```
```go
return &ACIInfo{
  Im: im,
  Key: key,
  Latest: latest,
  Time: t,
}
```

sgotti replied:
Dunno why I did this...

crawford commented on cas/cas.go:
```diff
@@ -36,6 +38,7 @@ import (
 const (
 	blobType int64 = iota
 	remoteType
+	ACIInfoType
```
Does this need to be exported?

sgotti replied:
definitely not! Fixed.

sgotti commented:
@crawford thanks for your review. I tried to fix your comments. Your comments were related to the patch proposed in #352 so I fixed it and rebased this PR on it (sorry for the confusion but this is due to the complexity and interactions of the various PRs proposed to reach #297).

Honestly I'll happily drop #352 and #353 (as this is a really ugly hack) in favor of something db based like discussed in #71 and #362 (with a proposed implementation based on ql in #361). (one doubt is related on the place to save the imagemanifest: in a diskv store or inside a blob column in the db table)

If you all can take a look at the above proposals so we can also avoid double work I'll be very happy.

crawford commented on cas/aciinfo.go:
```diff
@@ -0,0 +1,47 @@
+// Latest defines if the ACI was imported using the latest pattern (no version
+// label provided on ACI discovery)
+type ACIInfo struct {
+	Im      *schema.ImageManifest
```
The description is great, but I was suggestion an expansion of the struct member name. Maybe something like "ImageManifest" or just "Manifest".

sgotti commented:
I'm going to replace this with #392.
